src/ComServerHelpers/Internal/BaseActivationFactoryWrapper.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using ComServerHelpers.Windows.Com;
+using Windows.Win32.Foundation;
+using static Windows.Win32.PInvoke;
+
+namespace ComServerHelpers.Internal;
+
+[GeneratedComClass]
+internal sealed partial class BaseActivationFactoryWrapper(BaseActivationFactory factory, ComWrappers comWrappers) : IActivationFactory
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "COM method, must not throw.")]
+    public unsafe HRESULT ActivateInstance(void** instance)
+    {
+        if (instance is null)
+        {
+            return HRESULT.E_INVALIDARG;
+        }
+
+        nint unknown = 0;
+        try
+        {
+            object managedInstance = factory.ActivateInstance();
+            factory.OnInstanceCreated(managedInstance);
+
+            unknown = comWrappers.GetOrCreateComInterfaceForObject(managedInstance, CreateComInterfaceFlags.None);
+            *instance = (void*)unknown;
+        }
+        catch (Exception e)
+        {
+            if (unknown != 0)
+            {
+                Marshal.Release(unknown);
+            }
+
+            WinRT.ExceptionHelpers.SetErrorInfo(e);
+            return (HRESULT)WinRT.ExceptionHelpers.GetHRForException(e);
+        }
+        return HRESULT.S_OK;
+    }
+
+    public unsafe HRESULT GetIids(uint* iidCount, Guid** iids)
+    {
+        if (iidCount is null || iids is null)
+        {
+            return HRESULT.E_POINTER;
+        }
+
+        *iidCount = 1;
+        *iids = (Guid*)NativeMemory.Alloc((nuint)sizeof(Guid));
```

Comment:
What allocator does the callee expect for this to be allocated with? I'd be surprised if it expects it to be allocated with malloc/free.